### PR TITLE
[Merged by Bors] - feat: add standard Borel space class

### PR DESCRIPTION
**Pull request by Felix-Weilacher**
add a class `StandardBorelSpace`, whose relationship to `PolishSpace` is analogous to the relationship between `PolishSpace` and complete metric space. The usefulness is shown by `_root_.MeasurableSet.standardBorel`: A measurable subset of a standard Borel space is Polish. Note that a borel measurable subset of a Polish space will not typically be Polish in the induced topology!

Specific things done in this PR:

- added some basic instances for `StandardBorelSpace`
- changed theorems with hypotheses `[MeasurableSpace X] [TopologicalSpace X] [BorelSpace X] [PolishSpace X]` but which do not refer to the topology on `X` to depend instead on  `[MeasurableSpace X][StandardBorelSpace X]`. These automatically generalize the old theorems thanks to `standardBorel_of_polish`.

I believe this sort of refactoring can be pushed a lot farther in future PRs. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 1]
```diff
@@ -58,13 +58,86 @@ We use this to prove several versions of the Borel isomorphism theorem.
 
 open Set Function PolishSpace PiNat TopologicalSpace Metric Filter Topology MeasureTheory
 
-variable {α : Type*} [TopologicalSpace α] {ι : Type*}
+/-! ### Standard Borel Spaces -/
+
+variable (α : Type*)
+
+/-- A standard Borel space is a measurable space arising as the borel sets of some Polish topology.
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Constructions/Polish.lean:65]:
```suggestion
/-- A standard Borel space is a measurable space arising as the Borel sets of some Polish topology.
```

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 2]
```diff
@@ -58,13 +58,86 @@ We use this to prove several versions of the Borel isomorphism theorem.
 
 open Set Function PolishSpace PiNat TopologicalSpace Metric Filter Topology MeasureTheory
 
-variable {α : Type*} [TopologicalSpace α] {ι : Type*}
+/-! ### Standard Borel Spaces -/
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Constructions/Polish.lean:61]: Could you mention this in the file-level docstring?

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 3]
```diff
@@ -58,13 +58,86 @@ We use this to prove several versions of the Borel isomorphism theorem.
...
+/-- A convenience class similar to `UpgradedPolishSpace`. No instance should be registered.
+Instead one should use `letI := upgradePolishSpace α`. -/
+class UpgradedStandardBorel extends MeasurableSpace α, TopologicalSpace α,
+  BorelSpace α, PolishSpace α
+
+/-- Use as `letI := upgradePolishSpace α` to endow a standard Borel space `α` with
+a compatible Polish topology.
+
+Warning: following this with `borelize α` will cause an error. Instead, one can
+rewrite with `eq_borel_upgradeStandardBorel α`. -/
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Constructions/Polish.lean:85]:
```suggestion
rewrite with `eq_borel_upgradeStandardBorel α`. 
TODO: fix the corresponding bug in `borelize`. -/
```

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 4]
```diff
@@ -58,13 +58,86 @@ We use this to prove several versions of the Borel isomorphism theorem.
...
+  constructor
+
+/-- The `MeasurableSpace α` instance on a `StandardBorelSpace` `α` is equal to
+the borel sets of `upgradeStandardBorel α`.-/
+theorem eq_borel_upgradeStandardBorel [MeasurableSpace α] [StandardBorelSpace α] :
+    ‹MeasurableSpace α› = @borel _ (upgradeStandardBorel α).toTopologicalSpace :=
+  @BorelSpace.measurable_eq _ (upgradeStandardBorel α).toTopologicalSpace _
+    (upgradeStandardBorel α).toBorelSpace
+
+variable {α} [MeasurableSpace α]
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Constructions/Polish.lean:99]: from this point on, `[MeasurableSpace α]` is active for the rest of the file, while it is not relevant later on. Could you open a `section` and only activate it inside this section?

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 5]
```diff
@@ -443,12 +517,12 @@ end MeasureTheory
...
   [TopologicalSpace Y] [T2Space Y] [MeasurableSpace Y] [OpensMeasurableSpace Y] [MeasurableSpace β]
 
-/-- If `f : X → Y` is a surjective Borel measurable map from a Polish space to a topological space
-with second countable topology, then the preimage of a set `s` is measurable if and only if the set
-is measurable.
+/-- If `f : X → Y` is a surjective Borel measurable map from a standard Borel space
+to a topological space with second countable topology, then the preimage of a set `s`
+is measurable if and only if the set is measurable.
 One implication is the definition of measurability, the other one heavily relies on `X` being a
 Polish space. -/
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Constructions/Polish.lean:527]:
```suggestion
One implication is the definition of measurability, the other one heavily relies on `X` being a
standard Borel space. -/
```

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 6]
```diff
@@ -842,37 +934,65 @@ theorem measurableSet_exists_tendsto [hγ : OpensMeasurableSpace γ] [Countable
 
 end MeasureTheory
 
+namespace StandardBorelSpace
+
+variable {α : Type*} [MeasurableSpace α] [StandardBorelSpace α]
+
+/-- If `s` is a meaurable set in a standard Borel space, there is a compatible Polish topology
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Constructions/Polish.lean:941]:
```suggestion
/-- If `s` is a measurable set in a standard Borel space, there is a compatible Polish topology
```

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 7]
```diff
@@ -886,8 +1006,8 @@ end PolishSpace
 
 namespace MeasureTheory
 
-variable (α)
-variable [MeasurableSpace α] [PolishSpace α] [BorelSpace α]
+variable (α : Type*)
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Constructions/Polish.lean:1009]: Why did you change that? There is already a variable alpha in scope, and you are introducing a second one -- use `#where` to see what is already defined at some place.

File: Mathlib/Topology/MetricSpace/Polish.lean [hunk 8]
```diff
@@ -125,6 +125,14 @@ instance sigma {ι : Type*} [Countable ι] {E : ι → Type*} [∀ n, Topologica
   inferInstance
 #align polish_space.sigma PolishSpace.sigma
 
+/-- The product of two Polish spaces is Polish. -/
+instance prod [TopologicalSpace α] [PolishSpace α] [TopologicalSpace β] [PolishSpace β] :
+    PolishSpace (α × β) :=
+  letI := upgradePolishSpace α
+  letI := upgradePolishSpace β
+  letI : MetricSpace (α × β) := Prod.metricSpaceMax
```
Comment (sgouezel) [inline Mathlib/Topology/MetricSpace/Polish.lean:133]:
```suggestion
```
No need for this, typeclass inference will find it by itself.

**Review by sgouezel — COMMENTED** [review]
Looks great, thanks a lot!

File: Mathlib/MeasureTheory/Constructions/Polish.lean (hunk 7 above)
Comment (Felix-Weilacher) [reply Mathlib/MeasureTheory/Constructions/Polish.lean]: Sorry about that. I was trying to get an `alpha` without a `TopologicalSpace` instance, but I see now that this isn't a good way to do that.

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 9]
```diff
@@ -842,37 +946,65 @@ theorem measurableSet_exists_tendsto [hγ : OpensMeasurableSpace γ] [Countable
...
+  infer_instance
+
+end StandardBorelSpace
+
 /-! ### The Borel Isomorphism Theorem -/
 
 namespace PolishSpace
 
-variable {β : Type*} [TopologicalSpace β] [PolishSpace α] [PolishSpace β]
+variable {α β : Type*}
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Constructions/Polish.lean:977]: Same thing: don't introduce a new type with the same name as a type which is already in scope.

Comment (Felix-Weilacher) [reply Mathlib/MeasureTheory/Constructions/Polish.lean]: should all be fixed now.

**Comment by sgouezel** [comment]
bors r+
Thanks!
